ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/alter/rewrite/AlterMaterializedViewRewriteAnalyzer.java

```diff
@@ -68,10 +68,12 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
     Table materializedViewTable = getTable(tableName, true);
 
     // One last test: if we are enabling the rewrite, we need to check that query
-    // only uses transactional (MM and ACID) tables
+    // only uses transactional (MM and ACID and Iceberg) tables
     if (rewriteEnable) {
       for (SourceTable sourceTable : materializedViewTable.getMVMetadata().getSourceTables()) {
-        if (!AcidUtils.isTransactionalTable(sourceTable.getTable())) {
+        Table table = new Table(sourceTable.getTable());
```

Comment:
nit: (for future) ideally it would be good to extend SourceTable itself to allow apis such as isNonNative() such that temporary objects don't need to be created.

Reply:
`SourceTable` and `metastore.api.Table` are thrift generated classes to transfer data only. These classes don't contain any functionality. The one I created here is a `ql.metadata.Table` instance which wraps a `metastore.api.Table` it contains both data and some functionality parse and persist data.